app/src/main/java/com/vero/tasky/core/presentation/navigation/AppRoot.kt

```diff
@@ -0,0 +1,57 @@
+package com.vero.tasky.core.presentation.navigation
+
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Scaffold
+import androidx.compose.material.Snackbar
+import androidx.compose.material.SnackbarHost
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.compose.rememberNavController
+import com.vero.tasky.MainViewModel
+import com.vero.tasky.core.presentation.components.LocalSnackbarHostState
+
+@Composable
+fun AppRoot() {
+    val context = LocalContext.current
+    val viewModel: MainViewModel = hiltViewModel()
+
+    val localSnackbarHostState = LocalSnackbarHostState.current
+    CompositionLocalProvider(
+        LocalSnackbarHostState provides localSnackbarHostState
+    ) {
+
+        LaunchedEffect(true) {
+            viewModel.uiEvent.collect { errorMessage ->
+                localSnackbarHostState.showSnackbar(context.getString(errorMessage))
+            }
+        }
+
+        Scaffold(
+            modifier = Modifier.fillMaxSize(),
+            snackbarHost = {
+                SnackbarHost(hostState = LocalSnackbarHostState.current) { data ->
+                    Snackbar(
+                        backgroundColor = Color.Red,
+                        snackbarData = data,
+                        contentColor = Color.White,
+                        modifier = Modifier.testTag("snackbar")
+                    )
+                }
+            },
+        ) { paddingValues ->
+            RootNavigation(
+                navController = rememberNavController(),
```

Comment:
This nav controller should be passed as a parameter to `AppRoot`, since otherwise you can't control the navigation for test cases. Imagine you'd want to test some interaction on event detail screen and photo detail screen, then you would need to initially navigate to the event detail screen which doesn't work if you don't have the nav controller reference in tests